subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -809,12 +810,19 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec implements Dire
             }
         """
 
-        expect:
-        succeeds("customTask")
+        when:
+        expectThatExecutionOptimizationDisabledWarningIsDisplayed(executer,
+            "Type 'CustomTask' property 'eagerMap' where key of nested map is of type 'java.lang.Integer'. " +
+                "Reason: Key of nested map must be of type 'String'.",
```

Comment:
❓ Should this show the possible solution of changing the key to `String`?

Reply:
This is interesting, I wonder why. @jvandort seems to have made that change, maybe he knows?

Reply:
It was like that before I made that change, I just moved where this the `renderMinimalInformationAbout` call was made. It looks like not writing solutions originally appeared in this commit: https://github.com/gradle/gradle/commit/ead019708aba43ab00cf6e05048cd9d1466a45ce
See `ValidateStep`: `renderMinimalInformationAbout` was originally called, but then was used to call `renderedMessage` which called the `renderMinimalInformationAbout` override which did not print solutions.

Reply:
Right, and a reasoning for not including the doc link nor solutions is given [here](https://github.com/gradle/gradle/commit/ead019708aba43ab00cf6e05048cd9d1466a45ce#diff-588d5bbfb9e0139a77ce80ee3d7125ef69495242faca2d794e06e1e890405246R54) in that commit:

>  // Because our deprecation warning system doesn't support multiline strings (bummer!) both in rendering
>  // **and** testing (no way to capture multiline deprecation warnings), we have to resort to removing details
>  // and rendering

If this is something we want to address, it should be a separate effort.